### PR TITLE
simplelru: introduce onAdd callback

### PR DESCRIPTION
**Pull request by amery**
as requested on #9

### CODE REVIEW

File: x/simplelru/lru.go [hunk 1]
```diff
@@ -115,7 +127,7 @@ func (m *LRU[K, T]) Get(key K) (T, time.Time, bool) {
 	return zero, time.Time{}, false
 }
 
-// prune removes entries if space is needed. It tries
+// Prune removes entries if space is needed. It tries
```
Comment (karasz) [inline x/simplelru/lru.go:130]: comment is exported, function is not....

Comment (amery) [reply x/simplelru/lru.go:130]: @karasz eh? you explicitly complained about this comment change when I unexported prune

Comment (amery) [reply x/simplelru/lru.go]: @karasz commit droped

**Comment by amery** [comment]
`x/simplelru/v0.0.3` tagged @karasz

**Pull request merged**